WIP/Source/Veazy/src/test/java/vn/edu/fu/veazy/core/service/UserServiceTest.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    @Test
+    public void testSaveUser() throws Exception {
+        userService.saveUser(user);
+		UserModel user = userService.findUserByUsername(username);
+        Assert.assertNotNull(user);
+    }
+
+    @Test
+    public void testSaveUser2() throws Exception {
+        RegisterForm form = new RegisterForm();
+        form.setUsername("user2");
+        form.setEmail("user2@email.com");
+        form.setPassword("user2");
+        userService.saveUser(form);
+		
+		UserModel user = userService.findUserByUsername("user2");
+        Assert.assertNotNull(user);
+    }
+
+    @Test
+    public void testFindUserById() throws Exception {
+        UserModel user = userService.findUserById(userId);
+        Assert.assertNotNull(user);
+    }
+
+    @Test
+    public void testFindUserById2() throws Exception {
+        UserModel user = userService.findUserById(-1);
+        Assert.assertNull(user);
+    }
+
+    @Test
+    public void testFindUserById3() throws Exception {
+        UserModel user = userService.findUserById(null);
+        Assert.assertNull(user);
+    }
+
+    @Test
+    public void testFindUserByEmail() throws Exception {
+        UserModel user = userService.findUserByEmail(email);
+        Assert.assertNotNull(user);
+    }
+
+    @Test
+    public void testFindUserByEmail2() throws Exception {
+        UserModel user = userService.findUserByEmail(null);
+        Assert.assertNull(user);
+    }
+
+    @Test
+    public void testFindUserByEmail3() throws Exception {
+        UserModel user = userService.findUserByEmail("");
+        Assert.assertNull(user);
+    }
+
+    @Test
+    public void testFindUserByEmail4() throws Exception {
+        UserModel user = userService.findUserByEmail("notexistemail.com");
+        Assert.assertNull(user);
+    }
+
+    @Test
+    public void testFindUserByUserName() throws Exception {
+        UserModel user = userService.findUserByUsername(username);
+        Assert.assertNotNull(user);
+    }
+
+    @Test
+    public void testFindUserByUserName2() throws Exception {
+        UserModel user = userService.findUserByUsername(null);
+        Assert.assertNull(user);
+    }
+
+    @Test
+    public void testFindUserByUserName3() throws Exception {
+        UserModel user = userService.findUserByUsername("");
+        Assert.assertNull(user);
+    }
+
+    @Test
+    public void testFindUserByUserName4() throws Exception {
+        UserModel user = userService.findUserByUsername("notExistUserName");
+        Assert.assertNull(user);
+    }
+
+    @Test
+    public void testFindAllUser() throws Exception {
+        List<UserModel> findAllUser = userService.findAllUser();
+        Assert.assertNotNull(findAllUser);
+        Assert.assertNotEquals(0, findAllUser.size());
+    }
+
+    @Test
+    public void testUpdateUser() throws Exception {
+        userService.update(user, form);
+		UserModel user = userService.findUserById(userId);
+		Assert.assertEquals("first", user.getFirstName());
```

Comment:
test 2 method khác nhau


On Fri, Aug 19, 2016 at 6:48 PM, Nguyễn Ngọc Minh <notifications@github.com>
wrote:

> In WIP/Source/Veazy/src/test/java/vn/edu/fu/veazy/core/
> service/UserServiceTest.java
> <https://github.com/sangnvus/201605JS01/pull/25#discussion_r75468482>:
>
> > +        UserModel user = userService.findUserByUsername("notExistUserName");
> > +        Assert.assertNull(user);
> > +    }
> > +
> > +    @Test
> > +    public void testFindAllUser() throws Exception {
> > +        List<UserModel> findAllUser = userService.findAllUser();
> > +        Assert.assertNotNull(findAllUser);
> > +        Assert.assertNotEquals(0, findAllUser.size());
> > +    }
> > +
> > +    @Test
> > +    public void testUpdateUser() throws Exception {
> > +        userService.update(user, form);
> > +		UserModel user = userService.findUserById(userId);
> > +		Assert.assertEquals("first", user.getFirstName());
>
> sao gioongs case dưới thế
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sangnvus/201605JS01/pull/25/files/442cb029a2ed3ce65a0c4e6062ce71d64bbc74d1#r75468482>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASfcY6XwXi--Z4y3TuzZpJmzu0Lwyedgks5qhZgLgaJpZM4JnxSY>
> .
>